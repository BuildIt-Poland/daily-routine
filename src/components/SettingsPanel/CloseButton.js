import React from 'react';
import PropTypes from 'prop-types';

import Hitbox from '../Hitbox';
import { Cross } from '../Icons';

function CloseButton({ onClick }) {
  return (
    <Hitbox aria-label="Close" onClick={onClick} data-testid="close-button">
      <Cross />
    </Hitbox>
  );
}

CloseButton.propTypes = {
  onClick: PropTypes.func
};

export default CloseButton;
