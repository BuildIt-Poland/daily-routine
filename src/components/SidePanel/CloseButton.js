import React from 'react';
import PropTypes from 'prop-types';

import Hitbox from '../Hitbox';
import { Cross } from '../Icons';

function CloseButton({ onClick }) {
  return (
    <Hitbox aria-label="Close" onClick={onClick}>
      <Cross />
    </Hitbox>
  );
}

CloseButton.propTypes = {
  onClick: PropTypes.func.isRequired
};

export default CloseButton;
