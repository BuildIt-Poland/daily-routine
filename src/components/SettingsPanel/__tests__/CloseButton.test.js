import React from 'react';
import { create } from 'react-test-renderer';

import CloseButton from '../CloseButton';

jest.mock('../../Icons', () => ({
  Cross: 'Cross'
}));

describe('COMPONENT - SettingsPanel CloseButton', () => {
  it('renders correctly', () => {
    const component = create(<CloseButton />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
