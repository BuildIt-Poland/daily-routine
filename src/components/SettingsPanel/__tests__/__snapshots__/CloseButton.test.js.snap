// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - SettingsPanel CloseButton renders correctly 1`] = `
.c0 {
  position: relative;
  border: 0;
  margin: 0;
  padding: 0;
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 4rem;
  min-height: 4rem;
  cursor: pointer;
  outline: 0;
}

<button
  aria-label="Close"
  className="c0"
  data-testid="close-button"
>
  <Cross />
</button>
`;
